File: fbtc-adm/src/app/components/shared/model/anuidade-tipo-publico.ts
import { ValorAnuidade } from './valor-anuidade';
// import { TipoPublicoValorAnuidadeDao } from './tipo-publico';
export class AnuidadeTipoPublico {

    anuidadeTipoPublicoId: number;
    anuidadeId: number;
    tipoPublicoId: number;
    valoresAnuidades: ValorAnuidade[];
}

export class AnuidadeTipoPublicoDao extends AnuidadeTipoPublico {
    nomeTipoPublico: string;
    codigo: string;
}

File: Fbtc-Api/Fbtc.Domain/Entities/AnuidadeTipoPublico.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fbtc.Domain.Entities
{
    public class AnuidadeTipoPublico
    {
        public int AnuidadeTipoPublicoId { get; set; }
        public int AnuidadeId { get; set; }
        public int TipoPublicoId { get; set; }

        public IEnumerable<ValorAnuidade> ValoresAnuidades { get; set; }
    }

    public class AnuidadeTipoPublicoDao : AnuidadeTipoPublico
    {
        public string NomeTipoPublico { get; set; }
        public string Codigo { get; set; }
    }
}
